### PR TITLE
Rover: proximity and avoidance fix

### PR DESCRIPTION
**Pull request by rmackay9**
This PR fixes a few issues found while setting up simple object avoidance to work with ROS. The fixes are:

- remove the proximity checks from the Rover arming checks.  For copters these proximity checks mostly so that the pilot is aware of objects on the frame itself that may be obstructing the proximity sensor.   These obstructions could stop the vehicle from being able to move in certain directions after takeoff.  With Rovers this problem is much less important.
- ROS/mavros can provide OBSTACLE_DISTANCE messages to the AP_Proximity library but the update rate (even on an NVidia TX2) seems to be relatively low and irratic so I've increased the timeout to 0.5sec.  There are of course better solutions (such as improving the performance somehow of the ROS/mavros portion) but in general, I think the 200ms we had before was too low.
- AC_Avoid bug fix when using the "stop" method.  When looking for the intersection point with a polygon boundary we need to include the "margin" or else we will often not find the intersection point until long after the vehicle should have started slowing down.
- AC_Fence change to *not* return the 1st point in the _boundary array which is the "return to" point (that copter never uses).  This fix resolves an issue introduced by [this PR that was just merged to master](https://github.com/ArduPilot/ardupilot/pull/8790).

The AP_Avoid and AC_Fence changes should be included in Copter-3.6 although most Copter users stick with the default "slide" object avoidance.

This stop avoidance using a 360 lidar has been tested on a real rover.  The polygon fence change has been tested on a Copter in SITL.

### CODE REVIEW

File: libraries/AC_Avoidance/AC_Avoid.cpp [hunk 1]
```diff
@@ -435,7 +435,7 @@ void AC_Avoid::adjust_velocity_polygon(float kP, float accel_cmss, Vector2f &des
 
     // for stopping
     float speed = safe_vel.length();
-    Vector2f stopping_point = position_xy + safe_vel*((2.0f + get_stopping_distance(kP, accel_cmss, speed))/speed);
+    Vector2f stopping_point_plus_margin = position_xy + safe_vel*((2.0f + margin_cm + get_stopping_distance(kP, accel_cmss, speed))/speed);
```
Comment (peterbarker) [inline libraries/AC_Avoidance/AC_Avoid.cpp]: `const`

Comment (peterbarker) [reply libraries/AC_Avoidance/AC_Avoid.cpp]: Also, what's up with the magic 2cm here?

Comment (rmackay9) [reply libraries/AC_Avoidance/AC_Avoid.cpp]: @peterbarker, good question about the 2cm.  I don't really think we need it now that we've added the margin.  At the same time, it does no harm..

File: libraries/AP_Proximity/AP_Proximity_MAV.cpp [hunk 2]
```diff
@@ -21,6 +21,8 @@
 
 extern const AP_HAL::HAL& hal;
 
+#define PROXIMITY_MAV_TIMEOUT_MS    500 // distance messages must arrive at least every 0.5 seconds
```
[the remark this answers is not in the document]
Comment (rmackay9) [reply libraries/AP_Proximity/AP_Proximity_MAV.cpp]: ok, comment changed in follow up commit

File: libraries/AC_Fence/AC_Fence.cpp [hunk 3]
```diff
@@ -449,8 +449,12 @@ void AC_Fence::manual_recovery_start()
 /// returns pointer to array of polygon points and num_points is filled in with the total number
 Vector2f* AC_Fence::get_polygon_points(uint16_t& num_points) const
 {
-    num_points = _boundary_num_points;
-    return _boundary;
+    // return array minus the first point which holds the return location
+    num_points = MAX(_boundary_num_points - 1, 0);
```
[the remark this answers is not in the document]
Comment (rmackay9) [reply libraries/AC_Fence/AC_Fence.cpp]: great, thanks for finding this.  fixed in follow up commit.

**Pull request merged**